src/node/node_state.h

```diff
@@ -2051,6 +2051,14 @@ namespace ccf
             return std::make_unique<SerialisedMerkleTreeHook>(version, w);
           }));
 
+      network.tables->set_map_hook(
```

Comment:
What if the transaction is rolled back?

Reply:
Good point, we would need to be careful not to apply this too early. I think this should live outside of `raft.h`, but I don't know how to check for global commit outside of that. Would a hook in `Store::global_hooks` give us that guarantee?

Reply:
Yes, a global hook would do the job as its effects cannot be rolled back.